R/Writing R Code that Runs in Parallel.md

```diff
@@ -0,0 +1,159 @@
+
+`{dplyr}` is a powerful package for data manipulation in R, but it operates on a single core (or CPU). `{multidplyr}` extends `{dplyr}` by allowing operations to be performed in parallel across multiple cores, potentially reducing computation time.
+
+We will create a large dataset with 10 million rows and 256 numeric columns, then measure the time it takes to perform a data manipulation task using both `{dplyr}` and `{multidplyr}`.
+
+## Parallel Processing
+
+### What is Parallel Processing?
+
+Parallel processing is a method of computation where many calculations or processes are carried out simultaneously. Large problems can often be divided into smaller ones, which can then be solved at the same time. This is particularly useful for tasks that require significant computational power and time.
+
+### How Does Parallel Processing Relate to R?
+
+R is a single-threaded language by default, meaning it processes tasks sequentially, one after the other. This can be a limitation when working with large datasets or performing complex computations. The `{multidplyr}` package addresses this limitation by enabling parallel processing within the `{dplyr}` framework.
+
+`{multidplyr}` allows you to partition your data across multiple cores and perform `{dplyr}` operations in parallel. This can lead to significant performance improvements by utilising the full computational power of modern multi-core processors. By distributing the workload, `{multidplyr}` can reduce the time required for data manipulation tasks.
+
+### Why Use Parallel Processing?
+
+- **Speed**: By dividing tasks across multiple cores, parallel processing can significantly reduce the time required to complete large computations.
+- **Efficiency**: It allows for better utilisation of available computational resources, making it possible to handle larger datasets and more complex analyses.
+- **Scalability**: Parallel processing can scale with the number of available cores, making it suitable for both small and large-scale data processing tasks.
+
+### Reasons You Might Not Use Parallel Processing
+
+- **Overhead**: Setting up and managing parallel processes can introduce overhead, which might negate the performance benefits for smaller tasks.
+- **Complexity**: Writing and debugging parallel code can be more complex than writing sequential code.
+- **Memory Usage**: Parallel processing can increase memory usage, as each core may require its own copy of the data.
+
+### Inherently (or Embarrasingly) Parallel Data Manipulation Tasks
+
+Inherently (or Embarrassingly) parallel tasks are those that can be easily divided into independent subtasks, each of which can be processed simultaneously without requiring communication between the subtasks. This type of parallelism is particularly efficient because it minimises the overhead associated with inter-process communication.
```

Comment:
```suggestion
Inherently (or embarrassingly) parallel tasks are those that can be easily divided into independent subtasks, each of which can be processed simultaneously without requiring communication between the subtasks. This type of parallelism is particularly efficient because it minimises the overhead associated with inter-process communication.
```